packages/engine/src/simulation/package/run.rs

```diff
@@ -298,6 +300,9 @@ impl StepPackages {
 
         let collected = futs.collect::<Vec<_>>().await;
 
+        // Collect outputs and put back packages that we took.
+        // Output packages can't reload state batches, since they only have read access to state,
```

Comment:
Apologies it should be in `run_output_packages` (where the comment is right now doesn't make sense, will duplicate this comment on the updated commit and resolve this thread.)

Reply:
Yes I think that was the point of the comment originally. Output packages can't reload state as they don't have write access, and thus it has to be reloaded before the packages themselves are ran. (Which is why we do it at the top of `run_output_packages`